apm/src/app/user/user.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { IUser } from './user';
+import { UserService } from './user.service';
+
+@Component({
+  templateUrl: './user.component.html',
+  styleUrls: ['./user.component.css']
+})
+export class UserComponent implements OnInit {
+
+  public editForm!: FormGroup;
+  private firstName!: FormControl;
+  private lastName!: FormControl;
+  private homepageUrl = "/welcome";
+  constructor(private userService: UserService, private router: Router) { }
+  
+  ngOnInit(): void {
+    this.firstName = new FormControl(
+      this.userService.currentUser?.firstName, Validators.required
```

Comment:
`this.userService.currentUser?.firstName` can have a value of undefined potentially. I mean the code is also indicating towards that. you can use `this.userService.currentUser?.firstName ?? ''` instead and make sure you are assigning an empty string when the left expression evaluates into null or undefined

Alternatively you can also use `this.userService.currentUser?.firstName || ''` but that is not accurate

example
0 || 'abc'  gives 'abc' because 0 is falsy value
0 ?? 'abc' gives 0 because it is not null or undefined